apidump/apidump.go

```diff
@@ -641,11 +641,11 @@ func (a *apidump) Run() error {
 	if a.TargetIsRemote() {
 		{
 			// Record the first resource usage data slightly before the
-			// stats log delay to ensure we include usage data in the first
+			// telemetry interval log delay to ensure we include usage data in the first
 			// telemetry upload.
 			var delay time.Duration
-			if 0 < a.StatsLogDelay {
-				delay = time.Duration(math.Max(a.StatsLogDelay-5, 1)) * time.Second
+			if 0 < a.TelemetryInterval {
+				delay = time.Duration(math.Max(a.TelemetryInterval-5, 1)) * time.Second
```

Comment:
Same problem here.  If we are breaking the assumption that the statistics are on a regular schedule, what happens to the unpredictably-sent request?

There will be telemetry at time t=5 minutes.  But, the one sent in between 0 and 5 minutes will be missing some information?

Reply:
Hmm, valid point. In this case, we can omit the delay and record the first usage ASAP